File: go_files/Ex6.go
package main

import (
  "fmt"
)

type Motor struct {
  Tipo   string
  Potencia int
}

func (m Motor) Info() string {
  return fmt.Sprintf("Tipo: %s, Potência: %d HP", m.Tipo, m.Potencia)
}

type Carro struct {
  Marca string
  Modelo string
  Ano int
  Motor Motor 
}

func (c Carro) Info() string {
  return fmt.Sprintf("Marca: %s, Modelo: %s, Ano: %d, Motor: [%s]", c.Marca, c.Modelo, c.Ano, c.Motor.Info())
}

func main() {
  motor := Motor{
    Tipo: "V8",
    Potencia: 450,
  }

  carro := Carro{
    Marca: "Ford",
    Modelo: "Mustang",
    Ano: 2024,
    Motor: motor,
  }

  fmt.Println(carro.Info())
}

File: pyton_files/ex6.py


class Motor:
    def __init__(self, potencia):
        self.potencia = potencia

    def display_power(self):
        return f"Potência do motor: {self.potencia} CV"

class Car:
    def __init__(self, marca, modelo, ano, motor):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.motor = motor  

    def display_details(self):
        return (f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}, "
                f"{self.motor.display_power()}")


def main():
    motor_v8 = Motor(450)  
    carro1 = Car("Ford", "Mustang", 2022, motor_v8)  

    
    print(carro1.display_details())


if __name__ == "__main__":
    main()

